signer/universal/universal.go

```diff
@@ -91,40 +147,78 @@ func NewSigner(root Root, policy *config.Signing) (signer.Signer, error) {
 		s, err = remote.NewSigner(policy)
 	} else {
 		if policy.NeedsLocalSigner() && policy.NeedsRemoteSigner() {
-			// Currently we don't support a hybrid signer
-			return nil, cferr.New(cferr.PolicyError, cferr.InvalidPolicy)
-		}
-
-		if policy.NeedsLocalSigner() {
-			// shouldProvide indicates whether the
-			// function *should* have produced a key. If
-			// it's true, we should use the signer and
-			// error returned. Otherwise, keep looking for
-			// signers.
-			var shouldProvide bool
-			// localSignerList is defined in the
-			// universal_signers*.go files. These activate
-			// and deactivate signers based on build
-			// flags; for example,
-			// universal_signers_pkcs11.go contains a list
-			// of valid signers when PKCS #11 is turned
-			// on.
-			for _, possibleSigner := range localSignerList {
-				s, shouldProvide, err = possibleSigner(&root, policy)
-				if shouldProvide {
-					break
-				}
+			s, err = newUniversalSigner(root, policy)
+		} else {
+			if policy.NeedsLocalSigner() {
+				s, err = newLocalSigner(root, policy)
 			}
-
-			if s == nil {
-				err = cferr.New(cferr.PrivateKeyError, cferr.Unknown)
+			if policy.NeedsRemoteSigner() {
+				s, err = remote.NewSigner(policy)
 			}
 		}
+	}
+
+	return s, err
+}
 
-		if policy.NeedsRemoteSigner() {
-			s, err = remote.NewSigner(policy)
+// getMatchingProfile returns the SigningProfile that matches the profile passed.
+func (s *Signer) getMatchingProfile(profile string) (*config.SigningProfile, error) {
```

Comment:
This should return the default profile (s.policy.Default) when the profile string is empty.